sourcecode/report_calculator.py

```diff
@@ -0,0 +1,109 @@
+import calendar
+
+class ReportCalculator:
+    def calculate_report_year(weather_data: dict) -> dict:
+        """
+        Calculates Highest Temperature, Lowest Temperature
+        and Humidity.
+
+        @params
+        :weather_data dict: hold field names as key and their values.
+
+        @return
+        :dict: contains report data
+        """
+        report = {
+            'highest_temp': float('-inf'),
+            'highest_temp_date': "",
+            'lowest_temp_date': "",
+            'lowest_temp': float('inf'),
+            'humidity': float('-inf'),
+            'humidity_date': "",
+        }
+        max_temp =  float('-inf')
+        max_temp_date = ""
+        min_temp = float('inf')
+        min_temp_date = ""
+        max_humid =  float('-inf')
+        max_humid_date = ""
+
+        for weather_day_info in weather_data:
```

Comment:
Avoid words which are the name of data structures in variable naming.
`weather_data` --> `weather_records`